iio/iio_ad9361/iio_ad9361.c

```diff
@@ -0,0 +1,2586 @@
+							 ad9361_phy->current_tx_bw_Hz);
+		}
+	}
+	if(ret < 0) {
+		return ret;
+	}
+
+	return len;
+}
+
+/**
+ * set_rf_dc_offset_tracking_en().
+ * @device:	Physical instance of a iio_axi_dac device.
+ * @buf:	Value to be written to attribute.
+ * @len:	Length of the data in "buf".
+ * @channel:	Channel properties.
+ * Return: Number of bytes written to device, or negative value on failure.
+ */
+static ssize_t set_rf_dc_offset_tracking_en(void *device, char *buf, size_t len,
+		const struct iio_ch_info *channel)
+{
+	struct ad9361_rf_phy *ad9361_phy = (struct ad9361_rf_phy *)device;
+	int8_t en_dis = str_to_int32(buf);
+
+	if(en_dis < 0) {
+		return en_dis;
+	}
+	ad9361_phy->rfdc_track_en = en_dis ? 1 : 0;
+	if(!channel->ch_out) {
+		return ad9361_tracking_control(ad9361_phy, ad9361_phy->bbdc_track_en,
+					       ad9361_phy->rfdc_track_en, ad9361_phy->quad_track_en);
+	}
+
+	return -ENOENT;
+}
+
+/**
+ * set_sampling_frequency_available().
+ * @device:	Physical instance of a iio_axi_dac device.
+ * @buf:	Value to be written to attribute.
+ * @len:	Length of the data in "buf".
+ * @channel:	Channel properties.
+ * Return: Number of bytes written to device, or negative value on failure.
+ */
+static ssize_t set_sampling_frequency_available(void *device, char *buf,
+		size_t len,
+		const struct iio_ch_info *channel)
+{
+	/* This function doesn't have an equivalent function in ad9361,
+	 * and it should be implemented there first */
+
+	return -ENOENT;
+}
+
+/**
+ * set_quadrature_tracking_en().
+ * @device:	Physical instance of a iio_axi_dac device.
+ * @buf:	Value to be written to attribute.
+ * @len:	Length of the data in "buf".
+ * @channel:	Channel properties.
+ * Return: Number of bytes written to device, or negative value on failure.
+ */
+static ssize_t set_quadrature_tracking_en(void *device, char *buf, size_t len,
+		const struct iio_ch_info *channel)
+{
+	struct ad9361_rf_phy *ad9361_phy = (struct ad9361_rf_phy *)device;
+	int8_t en_dis = str_to_int32(buf);
+
+	if(en_dis < 0) {
+		return en_dis;
+	}
+	ad9361_phy->quad_track_en = en_dis ? 1 : 0;
+	if(!channel->ch_out) {
+		return ad9361_tracking_control(ad9361_phy, ad9361_phy->bbdc_track_en,
+					       ad9361_phy->rfdc_track_en, ad9361_phy->quad_track_en);
+	}
+
+	return -ENOENT;
+}
+
+/**
+ * set_sampling_frequency().
+ * @device:	Physical instance of a iio_axi_dac device.
+ * @buf:	Value to be written to attribute.
+ * @len:	Length of the data in "buf".
+ * @channel:	Channel properties.
+ * Return: Number of bytes written to device, or negative value on failure.
+ */
+static ssize_t set_sampling_frequency(void *device, char *buf, size_t len,
+				      const struct iio_ch_info *channel)
+{
+	struct ad9361_rf_phy *ad9361_phy = (struct ad9361_rf_phy *)device;
+	uint32_t sampling_freq_hz = srt_to_uint32(buf);
+	ssize_t ret = ad9361_set_rx_sampling_freq (ad9361_phy, sampling_freq_hz);
+
+	if (ret < 0)
+		return ret;
+
+	return len;
```

Comment:
Is this an issue that we return `len` without changing or using it in any way in the method?